[HAL] Add support for 64 bit us timestamp

pan- opened the pull request:
## Description
This patch add support of 64 bit us timestamp in the HAL ticker API. 
As a result it is possible to schedule events in more than 2147 seconds as it was previously the case. 
The class Ticker, Timer and Timeout takes advantage of this new possibility. 

## Status
**READY**


## Migrations
If this PR changes any APIs or behaviors, give a short description of what *API users* should do when this PR is merged.

NO

pan- committed:
[HAL] Ad primitives for long lived timestamp to the ticker API.

- A new 64 timestamp type has been added: us_timestamp_t.
- Changed type of timestamp in ticker_events iinto us_timestamp_t.
- Event queue now have a to store a current, absolute timestamp.
- Add alternative versions of ticker_insert_event and ticker_read which accept
  in input us_timestamp_t.
- Add documentation explaining the limitation and behavior of ticker_read and
  ticker_insert_event.

pan- committed:
[HAL] Add Tests for the hal ticker API.

pan- committed:
[HAL] Add support of 64 bits timestamp in ticker API implementation.

pan- committed:
[HAL] Ensure us_ticker and lp_ticker queue are correctly initialized.

pan- committed:
[Driver] Add API to support insertion of us_timestamp_t in TimerEvent.

pan- committed:
[Drivers] Let ticker support long us timestamp.

pan- committed:
[Driver] Improve timer precision for ms and seconds.

0xc0170 reviewed (commented):
+1 for the tests, extensive, have to have another look at them :-)

Just general comments.

pan- committed:
Timer: add a function to read the timer in a 64bit variable.

The Timer implementation has been refactored to take advantage of this
implementation as the base for all functions reading the time elapsed.

pan- committed:
ticker api: format code to meet mbed standards.

pan- committed:
ticker_api: Code clarification

* update_interrupt renamed into schedule_interrupt
* update_current_timestamp renamed into update_present_time
* ticker_event_queue_t::timestamp renamed into
* ticker_event_queue_t::present_time
* Fix doxygen comments in ticker_api.h
* Update comments internal comments in mbed_ticker_api.c

pan- committed:
ticker api: Schedule immediately event in the past.

pan- committed:
ticker test: fix code format.

pan- committed:
ticker tests: optimize RAM usage.

pan- committed:
ticker test: Fix data scope dependency.

pan- committed:
lp_ticker test: Capture completion time at completion point.

pan- committed:
TimerEvent: Use ticker_insert_event_us instead of the old API in
insert_absolute.

With this patch, event insertion should be more precise.

pan- committed:
Ticker NCS36510: Fix tests of the status register.

pan- committed:
lp ticker test: Fix timestamp update during deepsleep test.

pan- committed:
NCS36510 RTC driver: Fix driver.

* Initialization clear interrupt status
* Remove state in management of interrupt
* Handle timestamp in the past
* Handle current seconds, even if out of the relative timestamp.
* Simplify interrupt handling logic.

LDong-Arm commented on drivers/TimerEvent.h:
```diff
@@ -44,9 +44,12 @@ class TimerEvent {
     virtual void handler() = 0;
 
-    // insert in to linked list
+    // insert relative timestamp in to linked list
```
I guess this was meant to be "relative to the 32-bit cycle" rather than "relative current time". But later contributors assumed the later, and it's only addressed by #14005.
